File: Chaper three Java/GasMileage.java

import java.util.Scanner;

public class GasMileage {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        while (true) {
            double miles = getInput(input, "Enter miles driven (-1 to quit): ");
            if (miles == -1) {
                System.out.println("Exiting program.");
                break;
            }

            double gallons = getInput(input, "Enter gallons used: ");
            double mpg = calculateMPG(miles, gallons);
            displayResult(mpg);
        }

        input.close();
    }

    static double getInput(Scanner sc, String message) {
        System.out.print(message);
        return sc.nextDouble();
    }

    static double calculateMPG(double miles, double gallons) {
        if (gallons == 0) return 0;
        return miles / gallons;
    }


    static void displayResult(double mpg) {
        System.out.printf("Miles per gallon: %.2f%n%n", mpg);
    }
}

File: Chapter 3/Gas_Mileage.c
#include <stdio.h>
int gallonMileage(int miles, int gallons) {
    if (gallons == 0) {
        return 0; // Avoid division by zero
    }
    return miles / gallons;
}

int main() {
    int miles, gallons;
    printf("Enter miles driven: ");
    scanf("%d", &miles);
    printf("Enter gallons used: ");
    scanf("%d", &gallons);

    int mileage = gallonMileage(miles, gallons);
    if (gallons == 0) {
        printf("Gallons used cannot be zero.\n");
    } else {
        printf("Miles per gallon: %d\n", mileage);
    }

    return 0;
}